syntax = "proto3";

package my_if;

service GrpcViewer {
    rpc SayHello (GrpcViewerHelloRequest) returns (GrpcViewerHelloReply) {}
    rpc DoAction (GrpcViewerActionRequest) returns (GrpcViewerActionResponse) {}
}

// The request message containing the user's name.
message GrpcViewerHelloRequest {
    string name = 1;
}

// The response message containing the greetings
message GrpcViewerHelloReply {
    string message = 1;
}

message GrpcViewerActionRequest {
    string action = 1;
    int32 width = 2;
    int32 height = 3;
    int32 type = 4;
    bytes data = 5;
}
message GrpcViewerActionResponse {
    string action = 1;
    int32 width = 2;
    int32 height = 3;
    int32 type = 4;
    bytes data = 5; 
}

